src/hooks/web3/useImportToken.tsx

```diff
@@ -1,66 +1,69 @@
-import { useBrandedToast } from "hooks/chakra"
-import { useState } from "react"
-import { useToken } from "wagmi"
-import { Text } from "@chakra-ui/react"
+import { useMutation } from "@tanstack/react-query"
+import { fetchToken } from "@wagmi/core"
+import { cellarDataMap } from "data/cellarDataMap"
+import { MutationEventArgs } from "types/hooks"
+type Args = {
+  address: string
+  imageUrl?: string
+}
 
-export const useImportToken = (address: string) => {
-  const toast = useBrandedToast()
+type UseImportTokenArgs = MutationEventArgs<
+  Args,
+  Awaited<ReturnType<typeof fetchToken>>
+>
 
-  const [isLoading, setLoading] = useState(false)
-  const {
-    data: tokenData,
-    error,
-    isLoading: tokenLoading,
-  } = useToken({
+export const useImportToken = ({
+  onSuccess,
+  onError,
+  onMutate,
+}: UseImportTokenArgs = {}) => {
+  const doImportToken = async ({
     address,
-  })
-
-  const loading = tokenLoading || isLoading
-
-  const doImportToken = async () => {
+    imageUrl,
+  }: {
```

Comment:
Could you not pass your `Args` type instead of hardcoding types here? This is also slightly hard to parse. May be worth writing another type. Something along these lines

```ts
type DoImportToken = ({ address, imageUrl }: Args) = Promise<void>
```

I'm unsure if that's 100% the correct syntax, but that +

```ts
const doImportToken: DoImportToken = async ({ address, imageUrl }) = {...}
```

is much easier to read 🤓